### PR TITLE
Loss function for multilabel_soft_margin_loss in paddle frontend

### LINKED ISSUE #20797: multi_label_so

### PR DESCRIPTION
**Pull request by Indraneel99**
Close #20797

### CODE REVIEW

**Comment by Indraneel99** [comment]
Hi team,

Some checks are failing (2 in 128), does this mean I am missing edge cases? Please let me know where can i find crash log ( I am unable to find it 😟). 

I will update the formatting. Thanks

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_loss.py [hunk 1]
```diff
@@ -390,3 +390,48 @@ def test_paddle_margin_ranking_loss(
         margin=margin,
         reduction=reduction,
     )
+
+@handle_frontend_test(
+    fn_tree="paddle.nn.functional.multilabel_soft_margin_loss",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("float"),
+        num_arrays=2,
+        exclude_min=True,
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_loss.py:399]: I believe that we can add this if we set the bounds of both `max` and `min` values of arrays to be generated

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @Indraneel99 

I tried to run your tests locally and I'm getting the following errors
image

Would you check it out and the comments I left below🙂

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_loss.py [hunk 2]
```diff
@@ -390,3 +390,50 @@ def test_paddle_margin_ranking_loss(
...
+    on_device,
+    fn_tree,
+    backend_fw,
+    frontend,
+    test_flags,
+):
+    x_dtype, x = dtype_and_x
+    weight_dtype, weight = dtype_and_weight
+    helpers.test_frontend_function(
+        input_dtypes=[
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_loss.py:425]: This parameter is supposed to be a list, not a nested list, you can concatenate both `x_dtype` and `weight_dtype` lists

File: ivy/functional/frontends/paddle/nn/functional/loss.py [hunk 3]
```diff
@@ -230,3 +230,21 @@ def margin_ranking_loss(input, other, label, margin=0.0, reduction="mean", name=
     out = ivy.atleast_1d(out)
 
     return out
+
+@with_supported_dtypes({"2.5.1 and below": ("float32","float64")},"paddle")
+@to_ivy_arrays_and_back
+def multilabel_soft_margin_loss(input, label, weight=None, reduction='mean', name=None):
```
Comment (hirwa-nshuti) [inline ivy/functional/frontends/paddle/nn/functional/loss.py:236]: Function name is written incorrectly would you rename it to `multi_label_soft_margin_loss`

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_loss.py [hunk 4]
```diff
@@ -390,3 +390,50 @@ def test_paddle_margin_ranking_loss(
         margin=margin,
         reduction=reduction,
     )
+
+@handle_frontend_test(
+    fn_tree="paddle.nn.functional.multilabel_soft_margin_loss",
```
Comment (hirwa-nshuti) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_loss.py:395]: The function should also be `multi_label_soft_margin_loss`

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey I added a few comments to help you run tests successfully for you to debug the failing tests

**Comment by Indraneel99** [comment]
Hi @hirwa-nshuti , Thanks for the review and comments. I have changed the function name to multi_label_soft_margin_loss and updated the code. All Tests are passing locally, but it seems some checks are failing in GitHub. please help me with the failing checks. Thanks

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Would you also try tp fix the merge conflicts so I can be able to run the tests locally😊
Thanks

**Comment by Indraneel99** [comment]
> Would you also try tp fix the merge conflicts so I can be able to run the tests locally😊 Thanks

I have resolved the conflicts. Please look into it. Thanks ☺️

File: ivy/functional/frontends/paddle/nn/functional/loss.py [hunk 5]
```diff
@@ -315,3 +315,54 @@ def nll_loss(
     if ignore_index >= 0 and ignore_index < ivy.shape(input)[1]:
         output = output - loss[ignore_index] / den
     return output
+  
+  
+@with_supported_dtypes({"2.5.1 and below": ("float32", "float64")}, "paddle")
+@to_ivy_arrays_and_back
+def nll_loss(
```
Comment (hirwa-nshuti) [inline ivy/functional/frontends/paddle/nn/functional/loss.py:322]: This should be removed as it was redefined from above

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @Indraneel99 

Good job so far 😊
There are assertion errors in backends would you look into this🙂
Thanks

**Comment by Indraneel99** [comment]
Hey @hirwa-nshuti, Thanks for reviewing (sorry for the earlier mistakes 😢 ). I have updated the function and now all tests are passing locally. Please check it. Thanks 😄.

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @Indraneel99 

Sorry for getting back to you this late 🙂
Would you resolve these merge conflicts so that the CI check should run
Thanks!

**Comment by Indraneel99** [comment]
Hey @hirwa-nshuti . I have resolved the conflicts. Please check the code. Thanks.

**Review by hirwa-nshuti — APPROVED** [review]
Hey @Indraneel99 

All looks great to me 😊
Thanks for contributing

**Comment by Indraneel99** [comment]
Hey @hirwa-nshuti . Thanks for all the help 🙂.
